token/program-2022/src/state.rs

```diff
@@ -4,12 +4,50 @@ use crate::instruction::MAX_SIGNERS;
 use arrayref::{array_mut_ref, array_ref, array_refs, mut_array_refs};
 use num_enum::TryFromPrimitive;
 use solana_program::{
+    clock::Epoch,
     program_error::ProgramError,
     program_option::COption,
     program_pack::{IsInitialized, Pack, Sealed},
     pubkey::Pubkey,
 };
 
+/// Different kinds of accounts. Note that `Mint`, `Account`, and `Multisig` types
+/// are determined exclusively by the size of the account, and are not included in
+/// the account data. These `AccountType`s are only included if extensions have been
+/// initialized.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AccountType {
+    /// Marker for 0 data
+    Uninitialized,
+    /// Mint account with additional extensions
+    Mint,
+    /// Token holding account with additional extensions
+    Account,
+}
+
+impl Default for AccountType {
+    fn default() -> Self {
+        Self::Uninitialized
+    }
+}
+
+/// Extensions that can be applied to mints or accounts.  Mint extensions must only be
+/// applied to mint accounts, and account extensions must only be applied to holding
```

Comment:
```suggestion
/// applied to mint accounts, and account extensions must only be applied to token holding
```

Reply:
One of these days we'll come up with a better term than "account"...